src/command_modules/azure-cli-interactive/azclishell/threads.py

```diff
@@ -0,0 +1,56 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import threading
+
+from azclishell._dump_commands import DUMP_TABLE
+from azclishell.az_completer import initialize_command_table_attributes
+
+
+class ExecuteThread(threading.Thread):
+    """ thread for executing commands """
+    def __init__(self, func, args):
+        super(ExecuteThread, self).__init__()
+        self.args = args
+        self.func = func
+
+    def run(self):
+        self.func(self.args)
+
+
+class ProgressViewThread(threading.Thread):
```

Comment:
Class name is misleading. Class name should represent what it does not how it is used.